src/embed/init.ts

```diff
@@ -53,49 +53,62 @@ window['_dlo_rulesFromOpsTeam'] = [
 */
 
 function _dlo_collectRules(): any[] {
-  const results: any[] = [];
-  Object.getOwnPropertyNames(window).forEach((propName) => {
-    if (propName.startsWith('_dlo_rules') === false) return;
-
-    const prop = (window as { [key: string]: any })[propName];
-    if (Array.isArray(prop) === false) {
-      Logger.getInstance().warn(LogMessageType.RuleInvalid,
-        { property: prop, reason: 'Rules list must be an array' });
-      return;
-    }
-
-    prop.forEach((rule: any) => {
-      results.push(rule);
+  try {
+    const startTime = Date.now();
+    const results: any[] = [];
+    Object.getOwnPropertyNames(window).forEach((propName) => {
+      if (propName.startsWith('_dlo_rules') === false) return;
+
+      const prop = (window as { [key: string]: any })[propName];
+      if (Array.isArray(prop) === false) {
+        Logger.getInstance().warn(LogMessageType.RuleInvalid,
+          { property: prop, reason: 'Rules list must be an array' });
+        return;
+      }
+
+      prop.forEach((rule: any) => {
+        results.push(rule);
+      });
     });
-  });
-  return results;
+    Logger.getInstance().info(`DLO rules processing time (ms): ${startTime - Date.now()}`);
+    return results;
+  } catch (err) {
+    Logger.getInstance().error(`Error collecting DLO rules: ${err}`);
```

Comment:
I'm wondering if you can just re-use `LogMessageType.RuleRegistrationError`?  If you need something more particular to this failure, you could add a new field to `LogMessageType`.  The err itself could be passed on as doing this `Logger.getInstance().error(LogMessageType.RuleRegistrationError, { reason: err.message });`